docs/data/data-grid/rows/rows.md

```diff
@@ -95,9 +95,42 @@ If you need some rows to have different row heights this can be achieved using t
 > The grid bases on the referential value of these props to cache their values and optimize the rendering.
 >
 > ```tsx
-> const handleGetRowHeight = React.useCallback(() => { ... }, []);
+> const getRowHeight = React.useCallback(() => { ... }, []);
 >
-> <DataGridPro getRowHeight={handleGetRowHeight} />
+> <DataGridPro getRowHeight={getRowHeight} />
+> ```
+
+### Dynamic row height
+
+Instead of a fixed row height, you can let the grid calculate the height of each row based on its content.
+To do so, return `"auto`" on the function passed to the `getRowHeight` prop.
```

Comment:
```suggestion
Instead of a fixed row height, you can let the grid calculate the height of each row based on its content.
To do so, return `"auto`" on the function passed to the `getRowHeight` prop. Similar to the [variable row height](/x/react-data-grid/rows/#dynamic-row-height) returning `null` or `undefined` will default to using the `rowHeight` prop.
```

Something in that direction - to still say that you can have granularity and only enable dynamic row height for specific rows. Doing that will alow the density to work as expected for the remaining rows.